shared/setupBuild.gradle

```diff
@@ -0,0 +1,76 @@
+apply plugin: 'cpp'
+apply plugin: 'google-test-test-suite'
+apply plugin: 'edu.wpi.first.NativeUtils'
+
+apply from: "${rootDir}/shared/config.gradle"
+apply from: "${rootDir}/shared/javacommon.gradle"
+
+// Windows specific functionality to export all symbols from a binary automatically
+nativeUtils {
+    exportsConfigs {
+        "${nativeName}" {}
+    }
+}
+
+model {
+    components {
+        "${nativeName}"(NativeLibrarySpec) {
+            sources {
+                cpp {
+                    source {
+                        srcDirs 'src/main/native/cpp'
+                        include '**/*.cpp'
+                    }
+                    exportedHeaders {
+                        srcDirs 'src/main/native/include'
+                        include "**/*.h"
+                    }
+                }
+            }
+
+            if(project.hasProperty('includePhotonTargeting')) {
+                binaries.all {
+                    lib project: ':photon-targeting', library: 'photontargeting', linkage: 'shared'
+                }
+            }
```

Comment:
Can we pull this into photon-lib's build.gradle somehow? this feels like a violation of seperation of concerns. Maybe as something like this?
```
tasks.withType(NativeComponentSpec).each { binaries.all {
                    lib project: ':photon-targeting', library: 'photontargeting', linkage: 'shared'
                }
```